example/app/config.yml

```diff
@@ -81,17 +81,52 @@ default:
     subnets:
       - '{{AWS_SUBNET}}'
 
+  alarms:
+    CPUHigh:
+      alarm_action: ScaleOutScalingPolicy
+      threshold: '50'
+    CPULow:
+      alarm_action: ScaleInScalingPolicy
+      comparison_operator: LessThanThreshold
+      threshold: '50'
+
   ecs:
     instanceType: t2.medium
     desiredInstances: 1
+    maxInstances: 2
+    minInstances: 1
     availabilityZone: us-east-1b
     amiid: ami-a7a242da
     publicIp: true
     restartTasksOnDeploy: true
     docker:
       username: cumulususer
+    autoscaling_policies:
```

Comment:
Is this being applied to the EC2 instances, the ECS cluster services, or both?

Reply:
Both the alarms and the policies are being applied / evaluated against the ECS auto scaling group. The alarms are configured with an action, so when an alarm triggers, (in this case) it triggers an autoscaling policy activity. 

So in answer to your question - EC2 instances. Service auto scaling is possible but I think the service has to be set up as an application auto scaling  group which is not done yet - what do you think next steps are here?

Reply:
If we scale up EC2 instances but don't add any more service instances, has scaling actually happened?  We wouldn't actually be able to handle more load.  So yes, there is more that's needed.  We need service auto-scaling.  (Unless I'm grossly misunderstanding, which I'll always accept as a possibility)

Reply:
The workflow I've added runs tasks on ECS without using ECS services: It uses the native `ecs.runTask` state type step function integration which does not run using an ECS Service. I thought scaling using EC2 instances was sufficient given we wanted to limit the impact of this scaling given our expected use of Fargate in the future. There was no mention of scaling services in the ticket which is why I went with this implementation.

Based on this feedback, I would propose to close this PR and start again from the new understanding that scaling should be based on number of tasks and running services. I would just note that there are 2 ways to run ECS tasks from a step function: as an activity running within an ECS service or as a step function state using the new SF `ecs.runTask` integration. If we want to support scaling both we need scaling of both services and ec2 auto scaling groups.